### PR TITLE
Add headers to enable SharedArrayBuffer in stories

### PR DESCRIPTION
**Pull request by darthtrevino**
Issue:

`SharedArrayBuffer` is not available by default. In 2020, SharedArrayBuffer was reintroduced in some browsers provided that certain headers are present on the web assets. (See [the Mozilla Documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)). 

We have some stories in our [graph visualization library](https://github.com/microsoft/Graspologic-js) that perform graph layouts within a worker context, and these use a SharedArrayBuffer to efficiently communicate data. But right now those stories are broken because the memory is not shared across the worker context.

## What I did

This PR adds the required headers to the Storybook Dev Server to enable SharedArrayBuffer. It would be nice to be to configure custom headers in the Storybook configuration file, but that does not seem possible currently.

## How to test

I ran the storybook server locally and verified that SharedArrayBuffer was defined.

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
I see no reason NOT to do this..

**Comment by disambiguator** [comment]
Hi! Any chance that this will be merged? We're integrating some WASM into our project and are blocked by this change. In the meantime, any workaround available or way for us to locally patch this?

File: lib/core-server/src/dev-server.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ export async function storybookDevServer(options: Options) {
   app.use((req, res, next) => {
     res.header('Access-Control-Allow-Origin', '*');
     res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+    // These headers are required to enable SharedArrayBuffer
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer
+    res.header('Cross-Origin-Opener-Policy', 'same-origin');
```
Comment (rmnegatives) [inline lib/core-server/src/dev-server.ts:37]: @darthtrevino this caused me havoc on friday.
This exact example or any https:image file here got hit with cors.
```
// .storybook/YourTheme.js

import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'My custom storybook',
  brandUrl: 'https://example.com',
  brandImage: 'https://place-hold.it/350x150',
});
```
This also got hit with cors 
<script src="https://cdn.tailwindcss.com/"></script>

Error in chrome and firefox -> "Failed to load resource: net::ERR_BLOCKED_BY_RESPONSE.NotSameOriginAfterDefaultedToSameOriginByCoep"

I traced it to my yarn lock and then traced it from alpha 45 -> to alpha 50.

Comment (rmnegatives) [reply lib/core-server/src/dev-server.ts:37]: Overall not sure why but inside of storybook these headers stop getting attached. (Yes I see the config above.. looking into express)
access-control-allow-origin: *
access-control-expose-headers: *

Comment (rmnegatives) [reply lib/core-server/src/dev-server.ts:37]: If my understanding is right maybe you can add COOP and COEP as a flag for story book instead, just like ssl is a flag. That way if you need shared array buffer, you will have to deal with the reality that and external resource that does not support CORS will not work.. WHICH is working as expected. However IMO I find this to be more of an edge case and therefore fits better as a flag with some caveat text around it.

**Pull request merged**